js/main.js

```diff
@@ -14,11 +14,135 @@
   var MAX_COORDINATE_Y = 630;
   var PRICE = 10000;
 
-  // Скрываем лэйаут карты
+  // Вводим переменные
   var map = document.querySelector('.map');
-  map.classList.remove('map--faded');
-  var mapPinsArea = document.querySelector('.map__pins');
+  var mapPinsArea = map.querySelector('.map__pins');
   var pinTemplate = document.querySelector('#pin').content.querySelector('.map__pin');
+  var form = document.querySelector('.ad-form');
+  var formInputElements = form.querySelectorAll('input');
+  var formSelectElements = form.querySelectorAll('select');
+  var formSubmit = form.querySelector('button[type="submit"]');
+  var formTextarea = form.querySelector('textarea');
+  var mapFilters = map.querySelector('.map__filters');
+  var mainPin = map.querySelector('.map__pin--main');
+  var roomNumberValue = form.querySelector('#room_number');
+  var guestsNumber = form.querySelector('#capacity');
+  var validationMark = '';
+  var submitButton = form.querySelector('.ad-form__submit');
+
+  // Добавляем/убираем атрибут disabled у контролов
+  function controlsSetAttribute(controls) {
+    controls.forEach(function (control) {
+      control.setAttribute('disabled', 'disabled');
+    });
+  }
+
+  function controlsRemoveAttribute(controls) {
+    controls.forEach(function (control) {
+      control.removeAttribute('disabled');
+    });
+  }
+
+  setMapDefaultState();
+
+  // Задаем дефолтное состояние для карты и сразу же вызываем функцию
+  function setMapDefaultState() {
+    controlsSetAttribute(formInputElements);
+    controlsSetAttribute(formSelectElements);
+    mapFilters.setAttribute('disabled', 'disabled');
+    formTextarea.setAttribute('disabled', 'disabled');
+    formSubmit.setAttribute('disabled', 'disabled');
+  }
+
+  // Добавляем слушателя для инициализации карты, проверяем клик левой кнопкой
+
+  mainPin.addEventListener('mousedown', leftClickHandler);
+  mainPin.addEventListener('keydown', enterHandler);
+
+  function leftClickHandler(evt) {
+    if (evt.button === 0) {
+      initMapActiveState();
+    }
+  }
+
+  function enterHandler(e) {
+    if (e.key === 'Enter') {
+      initMapActiveState();
+    }
+  }
+
+  // Задаем координаты для поля адреса. Берем с помощью getBoundingRect значения по x,y, height и width пина.
+  // Добавляем значения острия, высотпу получаем из getComputedStyle
+  function setPinCoordinates() {
+    var pinCoordinates = mainPin.getBoundingClientRect();
+    var pinEdge = window.getComputedStyle(mainPin, ':after');
+    var pinEdgeHeight = parseInt(pinEdge.height, 10);
+    return 'x: ' + Math.floor(pinCoordinates.x + pinCoordinates.width / 2) + '; y: ' + Math.floor(pinCoordinates.y + pinCoordinates.height + pinEdgeHeight);
+  }
+
+  // Коллбэк, убирает класс с карты, рендерим пины, убираем атрибуты disabled, снимаем слушателя с mainPin
+  function initMapActiveState() {
+    var addressInput = document.querySelector('.ad-form__element > input#address');
```

Comment:
```suggestion
    var addressInput = document.querySelector('#address');
```
Красивый селект, но сложный, там у инпута айдишник